ctapipe/image/muon/muon_ring_finder.py

```diff
@@ -1,28 +1,139 @@
 import numpy as np
 import astropy.units as u
-from ctapipe.image.muon.ring_fitter import RingFitter
+from ctapipe.core import Component
 from ctapipe.io.containers import MuonRingParameter
+import traitlets as traits
+from iminuit import Minuit
+
+__all__ = ['MuonRingFitter']
+
+
+class MuonRingFitter(Component):
+    """Different ring fit algorithms for muon rings
+    """
+    fit_method = traits.CaselessStrEnum(
+        ['taubin', 'chaudhuri_kundu'],
+        default_value='chaudhuri_kundu'
+    ).tag(config=True)
+
+    tel_type = traits.CaselessStrEnum(
+        ['LST_LST_LSTCam', 'MST_MST_NectarCam', 'MST_MST_FlashCam', 'MST_SCT_SCTCam',
+         'SST_1M_DigiCam', 'SST_GCT_CHEC', 'SST_ASTRI_ASTRICam', 'SST_ASTRI_CHEC'],
+        default_value='LST_LST_LSTCam'
+    ).tag(config=True)
+
+    def __init__(self, config=None, parent=None, **kwargs):
+        super().__init__(config, parent, **kwargs)
+
+    def fitFormula(self, xc, yc, r):
+        '''taubin fit formula
+        reference : Barcelona_Muons_TPA_final.pdf (slide 6)
+        '''
+        upper_term = sum(
+            (
+                    (np.array(self.x) - xc) ** 2 +
+                    (np.array(self.y) - yc) ** 2
+                    - r ** 2
+            ) ** 2
+        )
+        lower_term = sum(
+            (
+                    (np.array(self.x) - xc) ** 2 +
+                    (np.array(self.y) - yc) ** 2
+            )
+        )
+
+        return np.abs(upper_term) / np.abs(lower_term)
+
+    def initialization(self):
+        """ initialization for taubin_fit with the
+        initial default values, errors, and limits
+        """
```

Comment:
Again, this should not hard code any starting values for a telescope. Where do those come from? 
You can calculate the most common r from the focal length and field of view for example, which you can get from the instrument description.

Reply:
If you need the FOV, i think the most clear way would be to use either the outer bounding circle of the camera, or perhaps an inner one avoiding the border (for odd-shaped cameras that may be better).  Something like:
```py3
cam = subarray.tel[x].camera
r_outer = np.hypot(cam.pix_x, cam.pix_y).max()
fov = np.pi * r_outer**2
```
or
```py3
m = cam.get_border_pixel_mask(2)   # move in 2 border layers
r_inner = np.hypot(cam[m].pix_x, cam[m].pix_y).max()
fov = np.pi * r_inner**2
```

THen you can convert that to degrees by using the focal length (`subarray.tel[x].optics.equivalent_focal_length`)

Reply:
I think regarding the initial values, we should enable the MuonRingFinder to configure the initial values of the respective method it uses via traitlets. At least that would be nice. 

But honestly, even nicer would be if the intial values would not matter too much, then we can fix them and just not bother. I'd like to study the behaviour of the fit from the initial values a little ... and see how much they matter. What do you think?

Reply:
We did not really solve the issue with the hardcoded stuff in `muon_reco_functions.py` yet, but I think in this PR we also did not make it worse. so I go ahead and resolve this comment.